interfaces/builtin/fwupd.go

```diff
@@ -64,29 +77,130 @@ const fwupdPermanentSlotAppArmor = `
   /sys/firmware/efi/esrt/entries/** r,
 
   # Allow fwupd to access system information
-  /sys/devices/virtual/dmi/id/product_name r,
-  /sys/devices/virtual/dmi/id/sys_vendor r,
+  /sys/firmware/dmi/tables/smbios_entry_point r,
+  /sys/devices/virtual/dmi/id/{sys,bios,product,chassis,board}_{name,version,vendor,date,serial,sku,type,asset_tag,family} r,
+
+  /sys/bus/usb/devices/ r,
+
+  /sys/firmware/acpi/bgrt/{type,version,{x,y}offset,image} r,
+
+  # Lenovo Think LMI plugin
+  /sys/devices/virtual/firmware-attributes/** r,
 
   # Allow read/write access for efivarfs filesystem
   /sys/firmware/efi/efivars/ r,
   /sys/firmware/efi/efivars/** rw,
 
+  /proc/modules r,
+  /run/udev/data/* r,
+  /sys/devices/**/usb*/busnum r,
```

Comment:
Fair. I will change that.

Reply:
I have removed all of the paths that are in `hardware-observer` from this PR. I have not removed the ones that were already here since it could break old setup using `uefi-fw-tools` snap.